### PR TITLE
Add unit tests for Utf8Span, react to Utf8String changes

### PR DESCRIPTION
**Pull request by GrabYourPitchforks**
Ref asm + unit tests for https://github.com/dotnet/coreclr/pull/26711. This PR will not pass CI until that build makes its way from coreclr to corefx.

### CODE REVIEW

File: src/System.Utf8String.Experimental/tests/System/Utf8SpanTests.Comparison.cs [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+            AssertEqualOrdinal(Utf8Span.Empty, Utf8Span.Empty);
+            AssertEqualOrdinal(Utf8Span.Empty, u8(""));
+        }
+
+        [Theory]
+        [InlineData(null, null, StringComparison.OrdinalIgnoreCase, null, true)]
+        [InlineData("encyclopaedia", "encyclopædia", StringComparison.OrdinalIgnoreCase, null, false)]
+        [InlineData("encyclopaedia", "encyclopædia", StringComparison.InvariantCulture, null, true)]
+        [InlineData("encyclopaedia", "ENCYCLOPÆDIA", StringComparison.InvariantCulture, null, false)]
+        [InlineData("encyclopaedia", "encyclopædia", StringComparison.InvariantCultureIgnoreCase, null, true)]
```
Comment (tarekgh) [inline src/System.Utf8String.Experimental/tests/System/Utf8SpanTests.Comparison.cs]: would be interesting adding a test cases for Greek sigma cases.

File: src/System.Utf8String.Experimental/tests/System/Utf8SpanTests.cs [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+
+            (int offset, int length) = range.GetOffsetAndLength(originalSpan.Bytes.Length);
+            Assert.True(Unsafe.AreSame(ref startOfSlicedSpan, ref Unsafe.Add(ref startOfOriginalSpan, offset)));
+            Assert.Equal(length, slicedSpan.Bytes.Length);
+        }
+
+        [Theory]
+        [InlineData("résumé", "2..")] // try to split the first 'é'
+        [InlineData("résumé", "..^1")] // try to split the last 'é'
+        public static void Indexer_ThrowsIfTryToSplitMultiByteSubsequence(string input, string rangeExpression)
```
Comment (tarekgh) [inline src/System.Utf8String.Experimental/tests/System/Utf8SpanTests.cs]: why this doesn't take range directly instead of expression string?

**Review by tarekgh — APPROVED** [review]

File: src/System.Utf8String.Experimental/tests/System/Utf8SpanTests.cs (hunk 2 above)
Comment (GrabYourPitchforks) [reply src/System.Utf8String.Experimental/tests/System/Utf8SpanTests.cs]: Only consts / primitives can be used as attribute data. :(
This was the most readable way I could think of to work around it.

**Comment by GrabYourPitchforks** [comment]
This went in as part of https://github.com/dotnet/corefx/pull/41280.

**Pull request closed without merging**